File: ServiceStack/tests/NorthwindAuto/wwwroot/dtos/dtos.vb
' Options:
'Date: 2024-10-17 23:05:22
'Version: 8.41
'Tip: To override a DTO option, remove "''" prefix before updating
'BaseUrl: http://localhost:20000
'
'''GlobalNamespace: 
'''MakePartial: True
'''MakeVirtual: True
'''MakeDataContractsExtensible: False
'''AddReturnMarker: True
'''AddDescriptionAsComments: True
'''AddDataContractAttributes: False
'''AddIndexesToDataMembers: False
'''AddGeneratedCodeAttributes: False
'''AddResponseStatus: False
'''AddImplicitVersion: 
'''InitializeCollections: True
'''ExportValueTypes: False
'''IncludeTypes: 
'''ExcludeTypes: 
'''AddNamespaces: 
'''AddDefaultXmlNamespace: http://schemas.servicestack.net/types

Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.Runtime.Serialization
Imports ServiceStack
Imports ServiceStack.DataAnnotations
Imports ServiceStack.Jobs

Namespace Global

    Namespace ServiceStack.Jobs

        Public Partial Class AdminCancelJobs
            Implements IReturn(Of AdminCancelJobsResponse)
            Implements IGet
            Public Sub New()
                Ids = New List(Of Long)
            End Sub

            Public Overridable Property Ids As List(Of Long)
            Public Overridable Property Worker As String
        End Class

        Public Partial Class AdminCancelJobsResponse
            Public Sub New()
                Results = New List(Of Long)
                Errors = New Dictionary(Of Long, String)
            End Sub

            Public Overridable Property Results As List(Of Long)
            Public Overridable Property Errors As Dictionary(Of Long, String)
            Public Overridable Property ResponseStatus As ResponseStatus
        End Class

        Public Partial Class AdminGetJob
            Implements IReturn(Of AdminGetJobResponse)
            Implements IGet
            Public Overridable Property Id As Nullable(Of Long)
            Public Overridable Property RefId As String
        End Class

        Public Partial Class AdminGetJobProgress
            Implements IReturn(Of AdminGetJobProgressResponse)
            Implements IGet
            <Validate(Validator:="GreaterThan(0)")>
            Public Overridable Property Id As Long

            Public Overridable Property LogStart As Nullable(Of Integer)
        End Class

        Public Partial Class AdminGetJobProgressResponse
            Public Overridable Property State As BackgroundJobState
            Public Overridable Property Progress As Nullable(Of Double)
            Public Overridable Property Status As String
            Public Overridable Property Logs As String
            Public Overridable Property DurationMs As Nullable(Of Integer)
            Public Overridable Property [Error] As ResponseStatus
            Public Overridable Property ResponseStatus As ResponseStatus
        End Class

        Public Partial Class AdminGetJobResponse
            Public Overridable Property Result As JobSummary
            Public Overridable Property Queued As BackgroundJob
            Public Overridable Property Completed As CompletedJob
            Public Overridable Property Failed As FailedJob
            Public Overridable Property ResponseStatus As ResponseStatus
        End Class

        Public Partial Class AdminJobDashboard
            Implements IReturn(Of AdminJobDashboardResponse)
            Implements IGet
            Public Overridable Property From As Nullable(Of Date)
            Public Overridable Property To As Nullable(Of Date)
        End Class

        Public Partial Class AdminJobDashboardResponse
            Public Sub New()
                Commands = New List(Of JobStatSummary)
                Apis = New List(Of JobStatSummary)
                Workers = New List(Of JobStatSummary)
                Today = New List(Of HourSummary)
            End Sub

            Public Overridable Property Commands As List(Of JobStatSummary)
            Public Overridable Property Apis As List(Of JobStatSummary)
            Public Overridable Property Workers As List(Of JobStatSummary)
            Public Overridable Property Today As List(Of HourSummary)
            Public Overridable Property ResponseStatus As ResponseStatus
        End Class

        Public Partial Class AdminJobInfo
            Implements IReturn(Of AdminJobInfoResponse)
            Implements IGet
            Public Overridable Property Month As Nullable(Of Date)
        End Class

        Public Partial Class AdminJobInfoResponse
            Public Sub New()
                MonthDbs = New List(Of Date)
                TableCounts = New Dictionary(Of String, Integer)
                WorkerStats = New List(Of WorkerStats)
                QueueCounts = New Dictionary(Of String, Integer)
            End Sub

            Public Overridable Property MonthDbs As List(Of Date)
            Public Overridable Property TableCounts As Dictionary(Of String, Integer)
            Public Overridable Property WorkerStats As List(Of WorkerStats)
            Public Overridable Property QueueCounts As Dictionary(Of String, Integer)
            Public Overridable Property ResponseStatus As ResponseStatus
        End Class

        Public Partial Class AdminQueryBackgroundJobs
            Inherits QueryDb(Of BackgroundJob)
            Implements IReturn(Of QueryResponse(Of BackgroundJob))
            Public Overridable Property Id As Nullable(Of Integer)
            Public Overridable Property RefId As String
        End Class

        Public Partial Class AdminQueryCompletedJobs
            Inherits QueryDb(Of CompletedJob)
            Implements IReturn(Of QueryResponse(Of CompletedJob))
            Public Overridable Property Month As Nullable(Of Date)
        End Class

        Public Partial Class AdminQueryFailedJobs
            Inherits QueryDb(Of FailedJob)
            Implements IReturn(Of QueryResponse(Of FailedJob))
            Public Overridable Property Month As Nullable(Of Date)
        End Class

        Public Partial Class AdminQueryJobSummary
            Inherits QueryDb(Of JobSummary)
            Implements IReturn(Of QueryResponse(Of JobSummary))
            Public Overridable Property Id As Nullable(Of Integer)
            Public Overridable Property RefId As String
        End Class

        Public Partial Class AdminQueryRequestLogs
            Inherits QueryDb(Of RequestLog)
            Implements IReturn(Of QueryResponse(Of RequestLog))
            Public Overridable Property Month As Nullable(Of Date)
        End Class

        Public Partial Class AdminQueryScheduledTasks
            Inherits QueryDb(Of ScheduledTask)
            Implements IReturn(Of QueryResponse(Of ScheduledTask))
        End Class

        Public Partial Class AdminRequeueFailedJobs
            Implements IReturn(Of AdminRequeueFailedJobsJobsResponse)
            Public Sub New()
                Ids = New List(Of Long)
            End Sub

            Public Overridable Property Ids As List(Of Long)
        End Class

        Public Partial Class AdminRequeueFailedJobsJobsResponse
            Public Sub New()
                Results = New List(Of Long)
                Errors = New Dictionary(Of Long, String)
            End Sub

            Public Overridable Property Results As List(Of Long)
            Public Overridable Property Errors As Dictionary(Of Long, String)
            Public Overridable Property ResponseStatus As ResponseStatus
        End Class

        Public Partial Class BackgroundJob
            Inherits BackgroundJobBase
            Public Overridable Property Id As Long
        End Class

        Public Partial Class BackgroundJobBase
            Implements IMeta
            Public Sub New()
                Args = New Dictionary(Of String, String)
                Meta = New Dictionary(Of String, String)
            End Sub

            Public Overridable Property Id As Long
            Public Overridable Property ParentId As Nullable(Of Long)
            Public Overridable Property RefId As String
            Public Overridable Property Worker As String
            Public Overridable Property Tag As String
            Public Overridable Property BatchId As String
            Public Overridable Property Callback As String
            Public Overridable Property DependsOn As Nullable(Of Long)
            Public Overridable Property RunAfter As Nullable(Of Date)
            Public Overridable Property CreatedDate As Date
            Public Overridable Property CreatedBy As String
            Public Overridable Property RequestId As String
            Public Overridable Property RequestType As String
            Public Overridable Property Command As String
            Public Overridable Property Request As String
            Public Overridable Property RequestBody As String
            Public Overridable Property UserId As String
            Public Overridable Property Response As String
            Public Overridable Property ResponseBody As String
            Public Overridable Property State As BackgroundJobState
            Public Overridable Property StartedDate As Nullable(Of Date)
            Public Overridable Property CompletedDate As Nullable(Of Date)
            Public Overridable Property NotifiedDate As Nullable(Of Date)
            Public Overridable Property RetryLimit As Nullable(Of Integer)
            Public Overridable Property Attempts As Integer
            Public Overridable Property DurationMs As Integer
            Public Overridable Property TimeoutSecs As Nullable(Of Integer)
            Public Overridable Property Progress As Nullable(Of Double)
            Public Overridable Property Status As String
            Public Overridable Property Logs As String
            Public Overridable Property LastActivityDate As Nullable(Of Date)
            Public Overridable Property ReplyTo As String
            Public Overridable Property ErrorCode As String
            Public Overridable Property [Error] As ResponseStatus
            Public Overridable Property Args As Dictionary(Of String, String)
            Public Overridable Property Meta As Dictionary(Of String, String)
        End Class

        Public Partial Class BackgroundJobOptions
            Public Sub New()
                Args = New Dictionary(Of String, String)
            End Sub

            Public Overridable Property RefId As String
            Public Overridable Property ParentId As Nullable(Of Long)
            Public Overridable Property Worker As String
            Public Overridable Property RunAfter As Nullable(Of Date)
            Public Overridable Property Callback As String
            Public Overridable Property DependsOn As Nullable(Of Long)
            Public Overridable Property UserId As String
            Public Overridable Property RetryLimit As Nullable(Of Integer)
            Public Overridable Property ReplyTo As String
            Public Overridable Property Tag As String
            Public Overridable Property BatchId As String
            Public Overridable Property CreatedBy As String
            Public Overridable Property TimeoutSecs As Nullable(Of Integer)
            Public Overridable Property Timeout As Nullable(Of TimeSpan)
            Public Overridable Property Args As Dictionary(Of String, String)
            Public Overridable Property RunCommand As Nullable(Of Boolean)
        End Class

        Public Enum BackgroundJobState
            Queued
            Started
            Executed
            Completed
            Failed
            Cancelled
        End Enum

        Public Partial Class CompletedJob
            Inherits BackgroundJobBase
        End Class

        Public Partial Class FailedJob
            Inherits BackgroundJobBase
        End Class

        Public Partial Class HourSummary
            Public Overridable Property Hour As String
            Public Overridable Property Total As Integer
            Public Overridable Property Completed As Integer
            Public Overridable Property Failed As Integer
            Public Overridable Property Cancelled As Integer
        End Class

        Public Partial Class JobStatSummary
            Public Overridable Property Name As String
            Public Overridable Property Total As Integer
            Public Overridable Property Completed As Integer
            Public Overridable Property Retries As Integer
            Public Overridable Property Failed As Integer
            Public Overridable Property Cancelled As Integer
        End Class

        Public Partial Class JobSummary
            Public Overridable Property Id As Long
            Public Overridable Property ParentId As Nullable(Of Long)
            Public Overridable Property RefId As String
            Public Overridable Property Worker As String
            Public Overridable Property Tag As String
            Public Overridable Property BatchId As String
            Public Overridable Property CreatedDate As Date
            Public Overridable Property CreatedBy As String
            Public Overridable Property RequestType As String
            Public Overridable Property Command As String
            Public Overridable Property Request As String
            Public Overridable Property Response As String
            Public Overridable Property UserId As String
            Public Overridable Property Callback As String
            Public Overridable Property StartedDate As Nullable(Of Date)
            Public Overridable Property CompletedDate As Nullable(Of Date)
            Public Overridable Property State As BackgroundJobState
            Public Overridable Property DurationMs As Integer
            Public Overridable Property Attempts As Integer
            Public Overridable Property ErrorCode As String
            Public Overridable Property ErrorMessage As String
        End Class

        Public Partial Class RequestLog
            Implements IMeta
            Public Sub New()
                Headers = New Dictionary(Of String, String)
                FormData = New Dictionary(Of String, String)
                Items = New Dictionary(Of String, String)
                ResponseHeaders = New Dictionary(Of String, String)
                Meta = New Dictionary(Of String, String)
            End Sub

            Public Overridable Property Id As Long
            Public Overridable Property TraceId As String
            Public Overridable Property OperationName As String
            Public Overridable Property DateTime As Date
            Public Overridable Property StatusCode As Integer
            Public Overridable Property StatusDescription As String
            Public Overridable Property HttpMethod As String
            Public Overridable Property AbsoluteUri As String
            Public Overridable Property PathInfo As String
            Public Overridable Property Request As String
            <StringLength(Integer.MaxValue)>
            Public Overridable Property RequestBody As String

            Public Overridable Property UserAuthId As String
            Public Overridable Property SessionId As String
            Public Overridable Property IpAddress As String
            Public Overridable Property ForwardedFor As String
            Public Overridable Property Referer As String
            Public Overridable Property Headers As Dictionary(Of String, String)
            Public Overridable Property FormData As Dictionary(Of String, String)
            Public Overridable Property Items As Dictionary(Of String, String)
            Public Overridable Property ResponseHeaders As Dictionary(Of String, String)
            Public Overridable Property Response As String
            Public Overridable Property ResponseBody As String
            Public Overridable Property SessionBody As String
            Public Overridable Property [Error] As ResponseStatus
            Public Overridable Property ExceptionSource As String
            Public Overridable Property ExceptionDataBody As String
            Public Overridable Property RequestDuration As TimeSpan
            Public Overridable Property Meta As Dictionary(Of String, String)
        End Class

        Public Partial Class ScheduledTask
            Public Overridable Property Id As Long
            Public Overridable Property Name As String
            Public Overridable Property Interval As Nullable(Of TimeSpan)
            Public Overridable Property CronExpression As String
            Public Overridable Property RequestType As String
            Public Overridable Property Command As String
            Public Overridable Property Request As String
            Public Overridable Property RequestBody As String
            Public Overridable Property Options As BackgroundJobOptions
            Public Overridable Property LastRun As Nullable(Of Date)
            Public Overridable Property LastJobId As Nullable(Of Long)
        End Class

        Public Partial Class WorkerStats
            Public Overridable Property Name As String
            Public Overridable Property Queued As Long
            Public Overridable Property Received As Long
            Public Overridable Property Completed As Long
            Public Overridable Property Retries As Long
            Public Overridable Property Failed As Long
            Public Overridable Property RunningJob As Nullable(Of Long)
            Public Overridable Property RunningTime As Nullable(Of TimeSpan)
        End Class
    End Namespace
End Namespace

File: ServiceStack/tests/NorthwindAuto/wwwroot/dtos/dtos.fs
(* Options:
Date: 2024-10-17 23:05:22
Version: 8.41
Tip: To override a DTO option, remove "//" prefix before updating
BaseUrl: http://localhost:20000

//GlobalNamespace: 
//MakeDataContractsExtensible: False
//AddReturnMarker: True
//AddDescriptionAsComments: True
//AddDataContractAttributes: False
//AddIndexesToDataMembers: False
//AddGeneratedCodeAttributes: False
//AddResponseStatus: False
//AddImplicitVersion: 
//ExportValueTypes: False
//IncludeTypes: 
//ExcludeTypes: 
//InitializeCollections: True
//AddNamespaces: 
*)

namespace ServiceStack

open System
open System.Collections
open System.Collections.Generic
open System.Runtime.Serialization
open ServiceStack
open ServiceStack.DataAnnotations

    [<AllowNullLiteral>]
    type RequestLog() = 
        member val Id:Int64 = new Int64() with get,set
        member val TraceId:String = null with get,set
        member val OperationName:String = null with get,set
        member val DateTime:DateTime = new DateTime() with get,set
        member val StatusCode:Int32 = new Int32() with get,set
        member val StatusDescription:String = null with get,set
        member val HttpMethod:String = null with get,set
        member val AbsoluteUri:String = null with get,set
        member val PathInfo:String = null with get,set
        member val Request:String = null with get,set
        [<StringLength(Int32.MaxValue)>]
        member val RequestBody:String = null with get,set

        member val UserAuthId:String = null with get,set
        member val SessionId:String = null with get,set
        member val IpAddress:String = null with get,set
        member val ForwardedFor:String = null with get,set
        member val Referer:String = null with get,set
        member val Headers:Dictionary<String, String> = new Dictionary<String, String>() with get,set
        member val FormData:Dictionary<String, String> = new Dictionary<String, String>() with get,set
        member val Items:Dictionary<String, String> = new Dictionary<String, String>() with get,set
        member val ResponseHeaders:Dictionary<String, String> = new Dictionary<String, String>() with get,set
        member val Response:String = null with get,set
        member val ResponseBody:String = null with get,set
        member val SessionBody:String = null with get,set
        member val Error:ResponseStatus = null with get,set
        member val ExceptionSource:String = null with get,set
        member val ExceptionDataBody:String = null with get,set
        member val RequestDuration:TimeSpan = new TimeSpan() with get,set
        member val Meta:Dictionary<String, String> = new Dictionary<String, String>() with get,set

    [<AllowNullLiteral>]
    type JobStatSummary() = 
        member val Name:String = null with get,set
        member val Total:Int32 = new Int32() with get,set
        member val Completed:Int32 = new Int32() with get,set
        member val Retries:Int32 = new Int32() with get,set
        member val Failed:Int32 = new Int32() with get,set
        member val Cancelled:Int32 = new Int32() with get,set

    [<AllowNullLiteral>]
    type HourSummary() = 
        member val Hour:String = null with get,set
        member val Total:Int32 = new Int32() with get,set
        member val Completed:Int32 = new Int32() with get,set
        member val Failed:Int32 = new Int32() with get,set
        member val Cancelled:Int32 = new Int32() with get,set

    [<AllowNullLiteral>]
    type WorkerStats() = 
        member val Name:String = null with get,set
        member val Queued:Int64 = new Int64() with get,set
        member val Received:Int64 = new Int64() with get,set
        member val Completed:Int64 = new Int64() with get,set
        member val Retries:Int64 = new Int64() with get,set
        member val Failed:Int64 = new Int64() with get,set
        member val RunningJob:Nullable<Int64> = new Nullable<Int64>() with get,set
        member val RunningTime:Nullable<TimeSpan> = new Nullable<TimeSpan>() with get,set

    type BackgroundJobState =
        | Queued = 0
        | Started = 1
        | Executed = 2
        | Completed = 3
        | Failed = 4
        | Cancelled = 5

    [<AllowNullLiteral>]
    type JobSummary() = 
        member val Id:Int64 = new Int64() with get,set
        member val ParentId:Nullable<Int64> = new Nullable<Int64>() with get,set
        member val RefId:String = null with get,set
        member val Worker:String = null with get,set
        member val Tag:String = null with get,set
        member val BatchId:String = null with get,set
        member val CreatedDate:DateTime = new DateTime() with get,set
        member val CreatedBy:String = null with get,set
        member val RequestType:String = null with get,set
        member val Command:String = null with get,set
        member val Request:String = null with get,set
        member val Response:String = null with get,set
        member val UserId:String = null with get,set
        member val Callback:String = null with get,set
        member val StartedDate:Nullable<DateTime> = new Nullable<DateTime>() with get,set
        member val CompletedDate:Nullable<DateTime> = new Nullable<DateTime>() with get,set
        member val State:BackgroundJobState = new BackgroundJobState() with get,set
        member val DurationMs:Int32 = new Int32() with get,set
        member val Attempts:Int32 = new Int32() with get,set
        member val ErrorCode:String = null with get,set
        member val ErrorMessage:String = null with get,set

    [<AllowNullLiteral>]
    type BackgroundJobBase() = 
        member val Id:Int64 = new Int64() with get,set
        member val ParentId:Nullable<Int64> = new Nullable<Int64>() with get,set
        member val RefId:String = null with get,set
        member val Worker:String = null with get,set
        member val Tag:String = null with get,set
        member val BatchId:String = null with get,set
        member val Callback:String = null with get,set
        member val DependsOn:Nullable<Int64> = new Nullable<Int64>() with get,set
        member val RunAfter:Nullable<DateTime> = new Nullable<DateTime>() with get,set
        member val CreatedDate:DateTime = new DateTime() with get,set
        member val CreatedBy:String = null with get,set
        member val RequestId:String = null with get,set
        member val RequestType:String = null with get,set
        member val Command:String = null with get,set
        member val Request:String = null with get,set
        member val RequestBody:String = null with get,set
        member val UserId:String = null with get,set
        member val Response:String = null with get,set
        member val ResponseBody:String = null with get,set
        member val State:BackgroundJobState = new BackgroundJobState() with get,set
        member val StartedDate:Nullable<DateTime> = new Nullable<DateTime>() with get,set
        member val CompletedDate:Nullable<DateTime> = new Nullable<DateTime>() with get,set
        member val NotifiedDate:Nullable<DateTime> = new Nullable<DateTime>() with get,set
        member val RetryLimit:Nullable<Int32> = new Nullable<Int32>() with get,set
        member val Attempts:Int32 = new Int32() with get,set
        member val DurationMs:Int32 = new Int32() with get,set
        member val TimeoutSecs:Nullable<Int32> = new Nullable<Int32>() with get,set
        member val Progress:Nullable<Double> = new Nullable<Double>() with get,set
        member val Status:String = null with get,set
        member val Logs:String = null with get,set
        member val LastActivityDate:Nullable<DateTime> = new Nullable<DateTime>() with get,set
        member val ReplyTo:String = null with get,set
        member val ErrorCode:String = null with get,set
        member val Error:ResponseStatus = null with get,set
        member val Args:Dictionary<String, String> = new Dictionary<String, String>() with get,set
        member val Meta:Dictionary<String, String> = new Dictionary<String, String>() with get,set

    [<AllowNullLiteral>]
    type BackgroundJob() = 
        inherit BackgroundJobBase()
        member val Id:Int64 = new Int64() with get,set

    [<AllowNullLiteral>]
    type CompletedJob() = 
        inherit BackgroundJobBase()

    [<AllowNullLiteral>]
    type FailedJob() = 
        inherit BackgroundJobBase()

    [<AllowNullLiteral>]
    type BackgroundJobOptions() = 
        member val RefId:String = null with get,set
        member val ParentId:Nullable<Int64> = new Nullable<Int64>() with get,set
        member val Worker:String = null with get,set
        member val RunAfter:Nullable<DateTime> = new Nullable<DateTime>() with get,set
        member val Callback:String = null with get,set
        member val DependsOn:Nullable<Int64> = new Nullable<Int64>() with get,set
        member val UserId:String = null with get,set
        member val RetryLimit:Nullable<Int32> = new Nullable<Int32>() with get,set
        member val ReplyTo:String = null with get,set
        member val Tag:String = null with get,set
        member val BatchId:String = null with get,set
        member val CreatedBy:String = null with get,set
        member val TimeoutSecs:Nullable<Int32> = new Nullable<Int32>() with get,set
        member val Timeout:Nullable<TimeSpan> = new Nullable<TimeSpan>() with get,set
        member val Args:Dictionary<String, String> = new Dictionary<String, String>() with get,set
        member val RunCommand:Nullable<Boolean> = new Nullable<Boolean>() with get,set

    [<AllowNullLiteral>]
    type ScheduledTask() = 
        member val Id:Int64 = new Int64() with get,set
        member val Name:String = null with get,set
        member val Interval:Nullable<TimeSpan> = new Nullable<TimeSpan>() with get,set
        member val CronExpression:String = null with get,set
        member val RequestType:String = null with get,set
        member val Command:String = null with get,set
        member val Request:String = null with get,set
        member val RequestBody:String = null with get,set
        member val Options:BackgroundJobOptions = null with get,set
        member val LastRun:Nullable<DateTime> = new Nullable<DateTime>() with get,set
        member val LastJobId:Nullable<Int64> = new Nullable<Int64>() with get,set

    [<AllowNullLiteral>]
    type AdminJobDashboardResponse() = 
        member val Commands:ResizeArray<JobStatSummary> = new ResizeArray<JobStatSummary>() with get,set
        member val Apis:ResizeArray<JobStatSummary> = new ResizeArray<JobStatSummary>() with get,set
        member val Workers:ResizeArray<JobStatSummary> = new ResizeArray<JobStatSummary>() with get,set
        member val Today:ResizeArray<HourSummary> = new ResizeArray<HourSummary>() with get,set
        member val ResponseStatus:ResponseStatus = null with get,set

    [<AllowNullLiteral>]
    type AdminJobInfoResponse() = 
        member val MonthDbs:ResizeArray<DateTime> = new ResizeArray<DateTime>() with get,set
        member val TableCounts:Dictionary<String, Int32> = new Dictionary<String, Int32>() with get,set
        member val WorkerStats:ResizeArray<WorkerStats> = new ResizeArray<WorkerStats>() with get,set
        member val QueueCounts:Dictionary<String, Int32> = new Dictionary<String, Int32>() with get,set
        member val ResponseStatus:ResponseStatus = null with get,set

    [<AllowNullLiteral>]
    type AdminGetJobResponse() = 
        member val Result:JobSummary = null with get,set
        member val Queued:BackgroundJob = null with get,set
        member val Completed:CompletedJob = null with get,set
        member val Failed:FailedJob = null with get,set
        member val ResponseStatus:ResponseStatus = null with get,set

    [<AllowNullLiteral>]
    type AdminGetJobProgressResponse() = 
        member val State:BackgroundJobState = new BackgroundJobState() with get,set
        member val Progress:Nullable<Double> = new Nullable<Double>() with get,set
        member val Status:String = null with get,set
        member val Logs:String = null with get,set
        member val DurationMs:Nullable<Int32> = new Nullable<Int32>() with get,set
        member val Error:ResponseStatus = null with get,set
        member val ResponseStatus:ResponseStatus = null with get,set

    [<AllowNullLiteral>]
    type AdminRequeueFailedJobsJobsResponse() = 
        member val Results:ResizeArray<Int64> = new ResizeArray<Int64>() with get,set
        member val Errors:Dictionary<Int64, String> = new Dictionary<Int64, String>() with get,set
        member val ResponseStatus:ResponseStatus = null with get,set

    [<AllowNullLiteral>]
    type AdminCancelJobsResponse() = 
        member val Results:ResizeArray<Int64> = new ResizeArray<Int64>() with get,set
        member val Errors:Dictionary<Int64, String> = new Dictionary<Int64, String>() with get,set
        member val ResponseStatus:ResponseStatus = null with get,set

    [<AllowNullLiteral>]
    type AdminQueryRequestLogs() = 
        inherit QueryDb<RequestLog>()
        interface IReturn<QueryResponse<RequestLog>>
        member val Month:Nullable<DateTime> = new Nullable<DateTime>() with get,set

    [<AllowNullLiteral>]
    type AdminJobDashboard() = 
        interface IReturn<AdminJobDashboardResponse>
        interface IGet
        member val From:Nullable<DateTime> = new Nullable<DateTime>() with get,set
        member val To:Nullable<DateTime> = new Nullable<DateTime>() with get,set

    [<AllowNullLiteral>]
    type AdminJobInfo() = 
        interface IReturn<AdminJobInfoResponse>
        interface IGet
        member val Month:Nullable<DateTime> = new Nullable<DateTime>() with get,set

    [<AllowNullLiteral>]
    type AdminGetJob() = 
        interface IReturn<AdminGetJobResponse>
        interface IGet
        member val Id:Nullable<Int64> = new Nullable<Int64>() with get,set
        member val RefId:String = null with get,set

    [<AllowNullLiteral>]
    type AdminGetJobProgress() = 
        interface IReturn<AdminGetJobProgressResponse>
        interface IGet
        [<Validate(Validator="GreaterThan(0)")>]
        member val Id:Int64 = new Int64() with get,set

        member val LogStart:Nullable<Int32> = new Nullable<Int32>() with get,set

    [<AllowNullLiteral>]
    type AdminQueryBackgroundJobs() = 
        inherit QueryDb<BackgroundJob>()
        interface IReturn<QueryResponse<BackgroundJob>>
        member val Id:Nullable<Int32> = new Nullable<Int32>() with get,set
        member val RefId:String = null with get,set

    [<AllowNullLiteral>]
    type AdminQueryJobSummary() = 
        inherit QueryDb<JobSummary>()
        interface IReturn<QueryResponse<JobSummary>>
        member val Id:Nullable<Int32> = new Nullable<Int32>() with get,set
        member val RefId:String = null with get,set

    [<AllowNullLiteral>]
    type AdminQueryScheduledTasks() = 
        inherit QueryDb<ScheduledTask>()
        interface IReturn<QueryResponse<ScheduledTask>>

    [<AllowNullLiteral>]
    type AdminQueryCompletedJobs() = 
        inherit QueryDb<CompletedJob>()
        interface IReturn<QueryResponse<CompletedJob>>
        member val Month:Nullable<DateTime> = new Nullable<DateTime>() with get,set

    [<AllowNullLiteral>]
    type AdminQueryFailedJobs() = 
        inherit QueryDb<FailedJob>()
        interface IReturn<QueryResponse<FailedJob>>
        member val Month:Nullable<DateTime> = new Nullable<DateTime>() with get,set

    [<AllowNullLiteral>]
    type AdminRequeueFailedJobs() = 
        interface IReturn<AdminRequeueFailedJobsJobsResponse>
        member val Ids:ResizeArray<Int64> = new ResizeArray<Int64>() with get,set

    [<AllowNullLiteral>]
    type AdminCancelJobs() = 
        interface IReturn<AdminCancelJobsResponse>
        interface IGet
        member val Ids:ResizeArray<Int64> = new ResizeArray<Int64>() with get,set
        member val Worker:String = null with get,set

